service/policy/db/namespaces.go

```diff
@@ -386,3 +398,212 @@ func (c PolicyDBClient) RemovePublicKeyFromNamespace(ctx context.Context, k *nam
 		KeyId:       k.GetKeyId(),
 	}, nil
 }
+
+// validateRootCertificate validates that the PEM string is a valid PEM-encoded root certificate
+func validateRootCertificate(pemStr string) error {
+	// Check that the PEM string contains "BEGIN CERTIFICATE"
+	if !strings.Contains(pemStr, "BEGIN CERTIFICATE") {
+		return errors.Join(db.ErrInvalidCertificate, errors.New("invalid PEM format: must contain BEGIN CERTIFICATE marker"))
+	}
+
+	// Check that the PEM string contains newlines (proper PEM formatting)
+	if !strings.Contains(pemStr, "\n") {
+		return errors.Join(db.ErrInvalidCertificate, errors.New("invalid PEM format: must contain newlines"))
+	}
+
+	// Decode PEM block
+	block, _ := pem.Decode([]byte(pemStr))
+	if block == nil {
+		return errors.Join(db.ErrInvalidCertificate, errors.New("invalid PEM format: failed to decode PEM block"))
+	}
+
+	// Verify it's a CERTIFICATE type
+	if block.Type != "CERTIFICATE" {
+		return errors.Join(db.ErrInvalidCertificate, fmt.Errorf("invalid PEM type: expected CERTIFICATE, got %s", block.Type))
+	}
+
+	// Parse the certificate
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return errors.Join(db.ErrInvalidCertificate, fmt.Errorf("invalid certificate: not a valid X.509 certificate: %w", err))
+	}
+
+	// Verify it's a root certificate (self-signed)
```

Comment:
I don't see anywhere we are validating the certificate signature to be valid.  We need to add a call to `cert.CheckSignatureFrom(cert)` to validate the self signed signature and validate the cert has not been changed.

Reply:
Signature verification via cert.CheckSignatureFrom(cert)